rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java

```diff
@@ -306,14 +310,60 @@ protected void addTimeouts(List<String> nodesUrl) {
         }
     }
 
+    protected void startNodeImplWithRetries(final InetAddress host, final int nbNodes, int retries)
+            throws RMException {
+        final List<String> depNodeURLs = new ArrayList<>(nbNodes);
+        try {
+            startNodeImpl(host, nbNodes, depNodeURLs);
+        } catch (Exception e) {
+            logger.warn("Failed nodes deployment in host : " + host + ", retries left : " + retries);
+            if (isInfiniteRetries(retries) || retries > 1) {
```

Comment:
Let's assume that  `retries` initialized to 2:

1. retries=2 --> startNodeImplWithRetries with retries=1
2. retries=1 --> do nothing

You have retried 1 time whereas 2 is expected.